neqo-transport/src/cc/classic_cc.rs

```diff
@@ -375,8 +381,14 @@ impl<T: WindowAdjustment> CongestionControl for ClassicCongestionControl<T> {
         now: Instant,
         cc_stats: &mut CongestionControlStats,
     ) -> bool {
+        // State needs to be saved before `on_congestion_event` is called where it changes to
+        // `State::RecoveryStart`.
+        let in_slow_start = self.state.in_slow_start();
         let cwnd_reduced = self.on_congestion_event(largest_acked_pkt, now);
         if cwnd_reduced {
+            if in_slow_start {
+                cc_stats.slow_start_exited = true;
+            }
             cc_stats.congestion_events_ecn += 1;
         }
```

Comment:
```suggestion
        let was_in_slow_start = self.state.in_slow_start();
        let cwnd_reduced = self.on_congestion_event(largest_acked_pkt, now);
        if cwnd_reduced {
                cc_stats.slow_start_exited |= was_in_slow_start;
                cc_stats.congestion_events_ecn += 1;
        }
```
This is the "similar pattern" mentioned earlier.